destination/config_test.go

```diff
@@ -85,3 +86,24 @@ func TestConfig_Validate(t *testing.T) {
 		})
 	}
 }
+
+func TestConfig_ParseTopic_ErrorOnIndexNotFound(t *testing.T) {
+	is := is.New(t)
+	template := `{{ index .Metadata "topiccc" }}`
+
+	cfg := Config{Topic: template}
+	topic, getTopic, err := cfg.ParseTopic()
+	is.NoErr(err)
+
+	is.Equal(topic, "")
+
+	rec := sdk.Record{
+		Metadata: map[string]string{
+			"topic": "testtopic",
+		},
+	}
+	topic, err = getTopic(rec)
+	is.True(err != nil) // expected error on topic not found
```

Comment:
It would be good to check for the error itself.

Reply:
I'm checking for the contents of the error now, so that we don't mistake a template error, but I'm not sure whether checking the error contents is the best way to solve this issue.

I've thought about adding a `var ErrTopicNotFound = errors.New ...` and then check for its presence with `errors.Is`, but I don't know, is it too much @hariso ?

Reply:
I'm thinking that if we are adding ErrTopicNotFound just for the tests, it's probably a bit too much. So maybe checking the error contents isn't the best way to generally solve this issue, I believe that in our case and for this test it's still better.